src/geolocation/controllers/geocoderController.js

```diff
@@ -1,17 +1,23 @@
 import { geolocationService as geocoder } from '../services';
 
+let instance = null;
 class GeocoderController {
-  addressToCoordinates = (request, response) =>
-    geocoder
-      .addressToCoordinates(request.params.address)
-      .then(coordinates => response.json(coordinates))
-      .catch(error => response.json({ message: error.message }));
+  constructor(service) {
+    if (!instance) instance = this;
+    this.service = service;
+  }
 
-  coordinatesToAddress = (request, response) =>
-    geocoder
-      .coordinatesToAddress(request.params.lat, request.params.lon)
-      .then(address => response.json(address))
-      .catch(error => response.json({ message: error.message }));
+  addressToCoordinates = async function getCoordinates(request, response) {
```

Comment:
@v1adko , I chose this way because I lose context `this` when use async functions. I will think how I can rewrite it like you said.